cmd/kubeadm/app/apis/kubeadm/validation/validation.go

```diff
@@ -492,8 +491,11 @@ func ValidateSocketPath(socket string, fldPath *field.Path) field.ErrorList {
 		if !filepath.IsAbs(u.Path) {
 			return append(allErrs, field.Invalid(fldPath, socket, fmt.Sprintf("path is not absolute: %s", socket)))
 		}
-	} else if u.Scheme != kubeadmapiv1beta2.DefaultUrlScheme {
-		return append(allErrs, field.Invalid(fldPath, socket, fmt.Sprintf("URL scheme %s is not supported", u.Scheme)))
+	} else {
+		err = validateScheme(u.Scheme)
```

Comment:
Is there any reason to introduce a whole new func here? `DefaultUrlScheme` is already diversified between windows and linux.

Reply:
I didn't see that. Thanks for pointing that out! I reverted this file